reports/models.py

```diff
@@ -112,16 +113,20 @@ def generate_document(self):
         """
         Generate and save the document
         """
-
-        content = self._run_instance_method('generate')
-        name = self._run_instance_method('get_report_filename')
-
-        # Setting save to false to avoid hashed_upload_to raising an exception
-        # because of document not having an attached file.
-        self.document.save(name, content, save=False)
-        self.save()
-
-        report_generated.send(sender=self.__class__, report=self)
+        try:
+            name = self._run_instance_method('get_report_filename')
+            content = self._run_instance_method('generate')
+
+            # Setting save to false to avoid hashed_upload_to raising an exception
+            # because of document not having an attached file.
+            self.document.save(name, content, save=False)
+            self.save()
+
+            report_generated.send(sender=self.__class__, report=self)
+        except ReportSkipped as ex:
+            logger.info("The report was skipped")
+            logger.info(ex)
+            self.delete()
```

Comment:
let's not delete the entry, just 
- upload plain txt file, saying "Report generation was skipped: {reason}"
- don't send notifications
- reason is read from ex.reason, default value is "no data"